src/controllers/cartController.ts

```diff
@@ -1,144 +1,206 @@
         });
       }
-  
-      return res.status(200).json({ message: 'Success', cart });
-    } catch (error) {
-      console.error(error);
-      return res.status(500).json({ error: 'Internal server error' });
     }
-  };
-  
-
-  export const deleteFromCart = async (req: Request, res: Response) => {
-    try {
-      const { userId, productId } = req.body;
-  
-      if (!userId || !productId) {
-        return res.status(400).json({ error: 'Both userId and productId are required in the request body' });
+
+    return res.status(200).json({ message: 'Success', cartItems });
+  } catch (error) {
+    console.error(error);
+    return res.status(500).json({ error: 'Internal server error' });
+  }
+};
+
+
+export const addToCart = async (req: Request, res: Response) => {
+  try {
+    const { error } = cartsSchema.validate(req.body);
+
+    if (error) {
+      return res.status(400).json({ error: error.details[0].message });
+    }
+
+    const { userId, quantity, product_id } = req.body;
+
+    let cart = await CartsModel.findOne({ where: { user_id: userId, product_id } });
+
+    if (cart) {
+      const currentQuantity: number = cart.get('quantity') as number;
+
+      if (quantity >= currentQuantity) {
+        await cart.update({ quantity });
       }
-  
-      const cart = await CartsModel.findOne({
-        where: { user_id: userId, product_id: productId },
-      });
-  
-      if (!cart) {
-        return res.status(404).json({ error: 'Product not found in the user\'s cart' });
+    } else {
+      const product = await ProductsModel.findOne({ where: { id: product_id } });
+
+      if (!product) {
+        return res.status(404).json({ error: 'Product not found for the specified product ID' });
       }
-  
-      await CartsModel.destroy({
-        where: { user_id: userId, product_id: productId },
+
+      const product_name: string = product.get('name') as string;
+      const finalPrice: number = parseFloat(product.get('finalPrice') as string);
+
+      cart = await CartsModel.create({
+        user_id: userId,
+        product_id,
+        product_name,
+        finalPrice,
+        quantity,
       });
-  
-      return res.status(200).json({ message: 'Product removed from cart successfully' });
-    } catch (error) {
-      console.error(error);
-      return res.status(500).json({ error: 'Internal server error' });
     }
-  };
-  
-  
-  
-  export const clearCart = async (req: Request, res: Response) => {
-    try {
-      const userId = parseInt(req.params.id);
-  
-      if (isNaN(userId)) {
-        return res.status(400).json({ error: 'Invalid user ID' });
-      }
-  
-      await CartsModel.destroy({
-        where: { user_id: userId }, 
-      });
-  
-      return res.status(200).json({ message: 'Cart cleared successfully' });
-    } catch (error) {
-      console.error(error);
-      return res.status(500).json({ error: 'Internal server error' });
+
+    return res.status(200).json({ message: 'Success', cart });
```

Comment:
message has to be "success" s is small letter

Reply:
DONE